pkgs/top-level/python-packages.nix

```diff
@@ -6952,19 +6952,22 @@ in modules // {
     };
   };
 
-  lti = buildPythonPackage rec {
-    version = "0.4.0";
+  lti = let self' = (self.override {self = self';}) // {pytest = self.pytest_27;};
+  in buildPythonPackage rec {
```

Comment:
If some package needs pytest 2.7 instead of the current 2.9, then fix that package and set it to explicitly pytest_27.
pytest is typically just a buildInput and not a propagatedBuildInput so there won't be any problems with multiple versions of the same package.